src/Transport/HttpTransport.php

```diff
@@ -146,20 +157,40 @@ public function close(?int $timeout = null): PromiseInterface
      */
     private function cleanupPendingRequests(): void
     {
+        $requestGenerator = function (): \Generator {
+            foreach ($this->pendingRequests as $key => $request) {
+                yield $key => $this->httpClient->sendAsyncRequest($request);
+            }
+        };
+
```

Comment:
Why did you move this outside? Wouldn't this actively interrupt the cleanup at the first error?

Reply:
No, this is just a closure declaration so nothing gets executed. I've moved it outside just to reduce the visual clutter of the `try` block